airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -89,6 +94,9 @@ public class BufferedStreamConsumer extends FailureTrackingAirbyteMessageConsume
   private boolean hasStarted;
   private boolean hasClosed;
 
+  private Instant lastFlushEmission;
+  Duration BUFFER_FLUSH_FREQUENCY = Duration.ofMinutes(configs.getFlushBufferFrequencyInMinutes());
```

Comment:
nit: make this `private final` + add a package-private `@VisibleForTesting` constructor that accepts a flush freq argument?

(this field is _effectively_ final in production, so it would be nice to make that explicit)

even better - it might take some refactoring, but adding some sort of `accept(AirbyteMessage message, Instant now)` method would let you simulate any timestamps in the world, without needing to sleep at all in tests